atlas_commons/utils.py

```diff
@@ -83,6 +83,27 @@ def split_into_halves(
     return left_volume, right_volume
 
 
+def assign_hemispheres(annotation_path, halfway_offset: int = 0, hemispheres_path = None):
+    annotation = voxcell.VoxelData.load_nrrd(annotation_path)
+
+    hemispheres_volume = np.zeros_like(annotation.raw, dtype=np.dtype('u1'))
+    z_halfway = hemispheres_volume.shape[2] // 2 + halfway_offset
+    hemispheres_volume[..., :z_halfway] = 1 # right hemisphere
+    hemispheres_volume[..., z_halfway:] = 2 # left hemisphere
+    hemispheres_volume[annotation.raw == 0] = 0 # outside the brain
+
+    hemispheres = annotation
+    hemispheres.raw = hemispheres_volume
+
+    if not hemispheres_path:
```

Comment:
instead of handling file output here, I would return the instance, and let the caller decide how they can to deal w/ the name, missing paths, etc, etc.